akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEvent.scala

```diff
@@ -21,6 +21,12 @@ import scala.annotation.tailrec
 
 object ServerSentEvent {
 
+  sealed trait EventIdValue
+  case object ResetId extends EventIdValue
+  final case class Id(id: String) extends EventIdValue {
```

Comment:
I hear you, but as this is a library that might get used in performance critic environments we should be careful. Also, the spec doesn't really talk about resetting, does it. So I'd prefer a `val emptyId: Option[String] = Some("")` (lowercase, only use uppercase for compile time constants).